component/jsontypes.go

```diff
@@ -0,0 +1,28 @@
+package component
+
+import (
+	"encoding/json"
+	"time"
+)
+
+// ComponentInfo represents a component in flow.
+type ComponentInfo struct {
```

Comment:
To abstract flow api out I had to move this to a neutral spot and this seemed the most reasonable.